sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1309,6 +1309,21 @@ object CodeGenerator extends Logging {
   // bytecode instruction
   final val MUTABLESTATEARRAY_SIZE_LIMIT = 32768
 
+  // The java source code generated by whole codegen is compiled on the Driver side and sent to
+  // each Executor as the execution code to process data. This is very effective in processing
+  // large amounts of data in a distributed environment. However, in the test environment,
+  // because the amount of data is not large or not executed in parallel, the compilation time
+  // of these Java source code will become a major part of the entire test runtime. When
+  // running test cases, we summarize the total compilation time and output it to the execution
+  // log for easy analysis and view.
+  private val _compileTime = new LongAccumulator
+
+  // Visible for testing
+  def compileTime: Long = _compileTime.sum
```

Comment:
We should mention that it's time in nanoseconds

Reply:
OK. I will add it.